phys2bids/physio_obj.py

```diff
@@ -0,0 +1,403 @@
+        del(cls.ch_name[idx])
+        del(cls.units[idx])
+
+    def check_trigger_amount(cls, thr=2.5, num_timepoints_expected=0, tr=0):
+        """
+        Counts trigger points and corrects time offset in
+        the list representing time.
+
+        Input
+        -----
+        cls: :obj: `blueprint_input`
+            The object on which to operate
+        thr: float
+            Threshold to be used to detect trigger points.
+            Default is 2.5
+        num_timepoints_expected: int
+            Number of expected triggers (num of TRs in fMRI)
+        tr: float
+            The Repetition Time of the fMRI data.
+
+        Output
+        ------
+        cls.num_timepoints_found: int
+            Property of the `blueprint_input` class.
+            Contains the number of timepoints found
+            with the automatic estimation.
+
+        Outcome
+        -------
+        cls.timeseries:
+            The property `timeseries` is shifted with the 0 being
+            the time of first trigger.
+        """
+        print('Counting trigger points')
+        # Use first derivative of the trigger channel to find the TRs,
+        # comparing it to a given threshold.
+        trigger_deriv = np.diff(cls.timeseries[1])
+        timepoints = trigger_deriv > thr
+        num_timepoints_found = timepoints.sum()
+        time_offset = cls.timeseries[0][timepoints.argmax()]
+
+        if num_timepoints_expected:
+            print('Checking number of timepoints')
+            if num_timepoints_found > num_timepoints_expected:
+                timepoints_extra = (num_timepoints_found -
+                                    num_timepoints_expected)
+                print(f'Found {timepoints_extra} timepoints'
+                      'more than expected!\n'
+                      'Assuming extra timepoints are at the end '
+                      '(try again with a more conservative thr)')
+
+            elif num_timepoints_found < num_timepoints_expected:
+                timepoints_missing = (num_timepoints_expected -
+                                      num_timepoints_found)
+                print(f'Found {timepoints_missing} timepoints'
+                      'less than expected!')
+                if tr:
+                    print('Correcting time offset, assuming missing timepoints'
+                          ' are at the beginning (try again with '
+                          'a more liberal thr')
+                    time_offset -= (timepoints_missing * tr)
+                else:
+                    print('Can\'t correct time offset, (try again specifying '
+                          'tr or with a more liberal thr')
+
+            else:
+                print('Found just the right amount of timepoints!')
+
+        else:
+            print('Cannot check the number of timepoints')
+
+        cls.timeseries[0] -= time_offset
+        cls.num_timepoints_found = num_timepoints_found
+
+
+class blueprint_output():
```

Comment:
```suggestion
class BlueprintOutput:
```

Again, capital case is recommended for classes!